src/core/sign.ml

```diff
@@ -133,7 +149,12 @@ let link : t -> unit = fun sign ->
   sign.sign_unops := StrMap.map hn !(sign.sign_unops);
   sign.sign_binops := StrMap.map hn !(sign.sign_binops);
   StrMap.iter (fun _ (s, _) -> Tree.update_dtree s) !(sign.sign_symbols);
-  sign.sign_quants := SymSet.map link_symb !(sign.sign_quants)
+  sign.sign_quants := SymSet.map link_symb !(sign.sign_quants);
+  let ind =
+    (fun s -> { ind_cons = List.map link_symb (s.ind_cons)
+              ; ind_prop = link_symb (s.ind_prop) })
+  in
+  sign.sign_ind := SymMap.map ind !(sign.sign_ind)
```

Comment:
SymMap.map is not the appropriate function. After OCaml's doc: "map f m returns a map with same domain as m". But we need to change the domain as well. So we in fact need to build a new map:
```
let fn s i m = SymMap.add (link_sym s) (link_inductive i) m in
SymMap.fold fn !(sign.sign_ind) SymMap.empty
```

Reply:
Yes you right I think. Sorry.